piff/hyperpsf.py

```diff
@@ -0,0 +1,277 @@
+        if logger:
+            logger.debug("Done building PSF")
+        return psf
+
+    def fit(self, stars, model_keys=[], interp_keys=[], model_comparer=None, model_comparer_weights=None, model_init=None, interp_init=None, model_error=None, interp_error=None, model_limit=None, interp_limit=None, logger=None, skip_fit=False, **kwargs):
+        """Fit the model!
+        :param stars:                       A list of StarData instances.
+        :param model:                       A Model instance that defines how to model the individual PSFs
+                                            at the location of each star.
+        :param interp:                      An Interp instance that defines how to do the interpolation of the
+                                            data vectors (produced by model for each star).
+        :param model_keys, interp_keys:     A list that gives the attributes of model/interp that the params correspond to
+        :param model_comparer:              Model that gets applied to both stars and model_stars to compare success of model. If none is given, compare pixels.
+        :param model_comparer_weights:      Weights to apply to the params from model_comparer
+        :param model_init, interp_init:     A list that gives the initial values
+        :param model_error, interp_error:   A list that gives the initial step sizes
+        :param model_limit, interp_limit:   A list that gives the limits of the fit
+        :param logger:                      A logger object for logging debug info. [default: None]
+        :param skip_fit:                    If True, do not run migrad fit
+        :param kwargs:                      kwargs to pass to iminuit fitter
+        """
+        if logger:
+            logger.info("Start fitting HyperPSF using %s stars", len(stars))
+            logger.debug("Model Comparer is %s", model_comparer)
+            logger.debug("Model is %s", self.model)
+            logger.debug("Interp is %s", self.interp)
+            logger.debug("Model params are %s", model_keys)
+            logger.debug("Interp params are %s", interp_keys)
+        from iminuit import Minuit
+
+        # set up interior args for running the fit function
+        self._set_fit_func_kwargs(stars, model_keys, interp_keys, model_comparer, model_comparer_weights, model_init, interp_init, model_error, interp_error, model_limit, interp_limit)
+
+        # set up iminuit object
+        self.minuit_kwargs = {'throw_nan': False,
+                              'pedantic': True,
+                              'print_level': 0,
+                              'errordef': 1,
+                              }
+
+        # update minuit_kwargs from set_fit_func_kwargs
+        self.minuit_kwargs.update(self._fit_func_kwargs_minuit)
+
+        # update minuit kwargs from the kwargs
+        self.minuit_kwargs.update(kwargs)
+
+        self._logger = logger
+        self._minuit = Minuit(self._fit_func, **self.minuit_kwargs)
```

Comment:
I think this is likely to be problematic, since here you are treating the arguments as a dict via kwargs, and below you require them to be in a particular order (`p0`,`p1`,etc. or my alternate `args` syntax).  I don't think Python guarantees kwargs to be listed in any particular order.  It's probably deterministic, so maybe it's fine.  But it's definitely bad form.  If you want the arguments to be a list, you should just treat them that way from the start, not via a `kwargs` dict.